src/test/java/io/managed/services/test/rhoas/KafkaRhoasBasicTests.java

```diff
@@ -128,6 +131,12 @@ public void clean() {
             LOGGER.error("delete kafka instance error: ", t);
         }
 
+        try {
+            KafkaMgmtApiUtils.deleteKafkaByNameIfExists(kafkaMgmtApi, KAFKA_CREATE_INSTANCE_NAME);
```

Comment:
also please make sure that user is capable of deleting kafka instance in given moment. 
assuming operation will pass: (kafka instance owner is changed) old owner is not supposed to be able to delete kafka instance.

Reply:
I've been testing it manually and the results are the follows ones:

1. Login as primary user: 
```
$ rhoas whoami
ad-hoc-primary
```

2.  Create a Kafka instance:
```
$ rhoas kafka create --name fede-example
{
  "billing_model": "standard",
  "browser_url": "https://qaprodauth.console.redhat.com/application-services/streams/kafkas/cgr6ajqvomdotrjlltv0/details/dashboard",
  "cloud_provider": "aws",
  "created_at": "2023-04-12T08:01:19.729107475Z",
  "egress_throughput_per_sec": "100Mi",
  "href": "/api/kafkas_mgmt/v1/kafkas/cgr6ajqvomdotrjlltv0",
  "id": "cgr6ajqvomdotrjlltv0",
  "ingress_throughput_per_sec": "50Mi",
  "instance_type": "standard",
  "instance_type_name": "Standard",
  "kind": "Kafka",
  "max_connection_attempts_per_sec": 270,
  "max_data_retention_period": "P14D",
  "max_data_retention_size": {
    "bytes": 1073741824000
  },
  "max_partitions": 1500,
  "multi_az": true,
  "name": "fede-example",
  **"owner": "ad-hoc-primary"**,
  "reauthentication_enabled": true,
  "region": "us-east-1",
  "size_id": "x1",
  "status": "accepted",
  "total_max_connections": 9000,
  "updated_at": "2023-04-12T08:01:19.729107475Z"
}
```


3. Modify the instance owner to the secondary user:
`$ rhoas kafka update --owner ad-hoc-secondary --name fede-example -y`

4. Verify that the instance has been modified successfully:
```
$ rhoas kafka describe --name fede-example
{
  "billing_model": "standard",
  "browser_url": "https://qaprodauth.console.redhat.com/application-services/streams/kafkas/cgr6ajqvomdotrjlltv0/details/dashboard",
  "cloud_provider": "aws",
  "created_at": "2023-04-12T08:01:19.729107Z",
  "egress_throughput_per_sec": "100Mi",
  "href": "/api/kafkas_mgmt/v1/kafkas/cgr6ajqvomdotrjlltv0",
  "id": "cgr6ajqvomdotrjlltv0",
  "ingress_throughput_per_sec": "50Mi",
  "instance_type": "standard",
  "instance_type_name": "Standard",
  "kind": "Kafka",
  "max_connection_attempts_per_sec": 270,
  "max_data_retention_period": "P14D",
  "max_data_retention_size": {
    "bytes": 1073741824000
  },
  "max_partitions": 1500,
  "multi_az": true,
  "name": "fede-example",
  **"owner": "ad-hoc-secondary",**
  "reauthentication_enabled": true,
  "region": "us-east-1",
  "size_id": "x1",
  "status": "preparing",
  "total_max_connections": 9000,
  "updated_at": "2023-04-12T08:01:47.387172Z"
}
```


5. As the primary user (I've never logged with the secondary), request the deletion of the instance:
```
$ rhoas whoami
ad-hoc-primary
$ rhoas kafka delete --name fede-example
? Confirm the name of the instance you want to delete (fede-example): fede-example
Kafka instance "fede-example" is being deleted
```

6. Wait a couple minutes and list the instances, in order to verify that the instance has been deleted:
```
$ rhoas kafka list
No Kafka instances were found.
```

@ henryZrncik Is this a bug??

Reply:
@henryZrncik

Reply:
explanation is in PM.